File: go/46-darts/darts.go
package darts

import "math"

func Score(x, y float64) int {
	distance := math.Sqrt(x*x + y*y)

	if distance > 10 {
		return 0
	} else if distance > 5 {
		return 1
	} else if distance > 1 {
		return 5
	} else {
		return 10
	}
}

File: ruby/119-darts/darts.rb
class Darts
  def initialize(x, y)
    @x = x
    @y = y
  end

  def score
    distance = Math.sqrt(@x**2 + @y**2)

    if distance <= 1
      10
    elsif distance <= 5
      5
    elsif distance <= 10
      1
    else
      0
    end
  end
end
